torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -145,6 +146,67 @@ void AtenInitialize() {
 
 }  // namespace
 
+namespace xla_pre_autograd {
+
+struct MaxPool2dAutogradFunction
+    : public torch::autograd::Function<MaxPool2dAutogradFunction> {
+  static torch::Tensor forward(torch::autograd::AutogradContext* ctx,
+                               torch::Tensor self,
+                               torch::IntArrayRef kernel_size,
+                               torch::IntArrayRef stride,
+                               torch::IntArrayRef padding,
+                               torch::IntArrayRef dilation, bool ceil_mode) {
+    ctx->saved_data["kernel_size"] = kernel_size;
+    ctx->saved_data["stride"] = stride;
```

Comment:
I dunno how the saved_data map works, but array-ref are ... refs, so unless the saved_data map has special handling to make copies, will those be valid in the backward?

Reply:
Arrayref is converted to IValue in the saved map. I'd assume this is okay since 1) it passed test 2) torchvision is using it the same way :D https://github.com/pytorch/vision/blob/e89c4c0198fd8cbd11344564162c68771524b2d7/torchvision/csrc/PSROIPool.h#L83